core/src/main/java/com/github/steanky/element/core/context/ElementContext.java

```diff
@@ -1,111 +1,531 @@
+     * context's <i>root node</i>. This method catches {@link ElementException}s that are thrown when elements are
+     * provided, and relays them to the supplied consumer after the list has been iterated. This can allow certain
+     * elements to fail to load without preventing others from doing so.
      *
-     * @param path               the path key
+     * @param listPath           the path string pointing to the ConfigList
      * @param dependencyProvider the {@link DependencyProvider} used to provide dependencies
-     * @param <TElement>         the type of the element object
-     * @return the contextual element object
+     * @param cache              whether to prefer caching elements or not
+     * @param collectionSupplier the function used to create a {@link Collection} implementation based on a known size
+     * @param exceptionHandler   the consumer used to handle exceptions
+     * @param <TElement>         the type of element object
+     * @param <TCollection>      the type of collection
+     * @return a collection of provided element objects
      */
-    <TElement> @NotNull TElement provideAndCache(final @Nullable String path,
-            final @NotNull DependencyProvider dependencyProvider);
+    default @NotNull <TElement, TCollection extends Collection<TElement>> TCollection provideCollection(
+            final @NotNull String listPath, final @NotNull DependencyProvider dependencyProvider, final boolean cache,
+            final @NotNull IntFunction<? extends TCollection> collectionSupplier,
+            final @NotNull Consumer<? super ElementException> exceptionHandler) {
+        Objects.requireNonNull(listPath);
+        Objects.requireNonNull(dependencyProvider);
+        Objects.requireNonNull(collectionSupplier);
+        Objects.requireNonNull(exceptionHandler);
+
+        PathSplitter pathSplitter = pathSplitter();
+
+        Object[] ethylenePath = pathSplitter.splitPathKey(listPath);
+        String normalized = pathSplitter.normalize(listPath);
+
+        ConfigElement listElement = rootNode().getElement(ethylenePath);
+
+        if (listElement == null || !listElement.isList()) {
+            throw new ElementException("expected ConfigList at '" + normalized + "', found " + listElement);
+        }
+
+        ConfigList list = listElement.asList();
+        TCollection elementCollection = collectionSupplier.apply(list.size());
+
+        ElementException exception = null;
+        for (int i = 0; i < list.size(); i++) {
+            try {
+                elementCollection.add(provide(pathSplitter.append(normalized, i), dependencyProvider, cache));
+            } catch (ElementException e) {
+                if (exception == null) {
+                    exception = e;
+                } else {
+                    exception.addSuppressed(e);
+                }
+            }
+        }
+
+        if (exception != null) {
+            exceptionHandler.accept(exception);
```

Comment:
why use this instead of throw

Reply:
for example, if you load 10 elements, and 1 has bad config and errors out, you can (optionally) log the exception but still use the other 9 elements